packages/mui-base/CONTRIBUTING.md

```diff
@@ -0,0 +1,180 @@
+# Contributing
+
+## Creating a new hook
+
+### File structure
+
+When creating a new hook, follow the file structure found in other hooks' directories:
+
+Taking the imaginary `useAwesomeControl` as an example:
+
+- 📂 `useAwesomeControl`
+  - 📄 `index.ts` - aggregates the public exports from all the other files in the directory
+  - 📄 `useAwesomeControl.test.tsx` - unit tests
+  - 📄 `useAwesomeControl.spec.tsx` - (optional) type tests
+  - 📄 `useAwesomeControl.ts` - the implementation of the hook
+  - 📄 `useAwesomeControl.types.ts` - type definitions
+
+### Implementation
+
+Most Base UI hooks have a similar structure:
+
+1. [Parameters destructuring](#parameters-destructuring)
+2. Hook-specific logic
+3. [Event handler factory functions](#event-handler-factory-functions)
+4. [Slot props resolvers](#slot-props-resolvers)
+
+#### Parameters destructuring
+
+The parameters type must be called `[HookName]Parameters`.
+There are docs generation scripts that require this pattern.
```

Comment:
I think we need more explanation here and an example. Where and how should the parameters be destructured?

Reply:
We can likely even drop this, contributors don't need this information at this point. It's an overhead that will not make them implement the hook itself better.

Reply:
IMO it does no harm to have this piece of information - it could help someone find a problem when the docs are not generated properly. Plus, MUIers can also use this doc.